myproject/leave/templates/leave/approve.html

```diff
@@ -0,0 +1,239 @@
+    position:all;
+    font-family: arial, sans-serif;
+    font-size: 20px;
+    color: maroon;
+}
+textarea{
+    width: auto;
+    border: 3px solid #ccc;
+    box-sizing: border-box;
+    padding: 10px 15px;
+    -webkit-transition: 0.5s;
+    transition: 0.5s;
+    outline: none;
+}
+
+textarea:focus{
+	border: 3px solid maroon;
+
+}
+
+input[type=text] {
+    width: auto;
+    padding: 12px 20px;
+    margin: 8px 0;
+    box-sizing: border-box;
+    border: 3px solid #ccc;
+    -webkit-transition: 0.5s;
+    transition: 0.5s;
+    outline: none;
+}
+
+input[type=text]:focus {
+    border: 3px solid maroon;
+}
+
+.button {
+    background-color: maroon; /* Green */
+    border: none;
+    color: white;
+    text-align: center;
+    text-decoration: none;
+    display: inline-block;
+    font-size: 16px;
+    margin: 4px 2px;
+    cursor: pointer;
+}
+
+.button1 {padding: 10px 24px;}
+ul{
+    position:all;
+	font-family: arial, sans-serif;
+	color: maroon;
+}
+
+h4{
+	font-family: arial, sans-serif;
+	font-size: 14px
+}
+
+p{
+	font-family: arial, sans-serif;
+	color: maroon;
+}
+menu{
+	
+	font-family: arial, sans-serif;
+	color:maroon;
+}
+fieldset{
+	border:none;
+	width:100px;
+	margin:0px auto;
+}
+
+table {
+    font-family: arial, sans-serif;
+    border-collapse: collapse;
+    width: 85%;
+    height: 50%;
+    color:maroon;
+}
+
+td, th {
+    border: 1px solid cornsilk;
+    text-align: left;
+    font: 20px 5px;
+    padding: 8px;
+}
+
+/*tr:nth-child(even) {
+    background-color: ;
+}*/
+</style>
+<script type="text/javascript">
+	base_url =  'http://127.0.0.1:8000/leave/'; 
+    $(function(){
+        $.ajax({
+            type: 'GET',
+            url: base_url+'approval/naga/?format=json',
+            success: function(data){
+                $.each(data, function(index, data){
+                    var check_status = "Pending"
+                    if(data.approval_status == check_status){
+                			$("#employee_name").val(data.employee_name)
+	                    	$("#reporter").val(data.reporter)
+	                    	$("#from_date").val(data.from_date)
+	                    	$("#to_date").val(data.to_date)
+	                    	$("#no_days").val(data.no_days)
+	                    	$("#leave_type_name").val(data.leave_type_name)
+	                    	$("#reason").val(data.reason)
+	                    	$("#approval_status").val(data.approval_status)
+	                    }
+                    });
+
+                }
+            });
+        });
+        
+    $(function approve(){
+        $.ajax({
+            type: 'PUT',
+            url: base_url+'approval/naga/?format=json',
+            data: {
+                approval_status: 'Approved'
+            },
+            dataType: "json",
+            success: (result) => {
+                console.log("success")
+
+            	}
+        	});
+    	});
+	
+    	function approve(){
+            var dataForm = {approval_status='Approved'}
```

Comment:
Approval process needs to be done in the back end.
Without using the request Id how are you approving it?